pkg/controller/versions/strategy_v2_4.go

```diff
@@ -274,10 +274,41 @@ func (v *versionStrategyV2_4) ValidateRequest(ctx context.Context, cl client.Cli
 			return admission.ValidationResponse(false, "a cluster-scoped SMCP may only be created by users with cluster-admin permissions")
 		}
 	}
-
+	hasPilotSecretName := ContainsPilotSecretNameField(smcp)
+	if hasPilotSecretName {
+		return admission.ValidationResponse(false, "SMCP does not allow PilotSecretName field")
+}
+	hasRootCAConfigMapName := ContainsRootCAConfigMapNameField(smcp)
+	if hasRootCAConfigMapName {
+	return admission.ValidationResponse(false, "SMCP does not allow RootCAConfigMapName field")
+}
```

Comment:
I noticed you add this validation code to the `ValidateRequest` function, which isn't the best place for it. Instead, I think you should create a `validateSecurity` function and call it in `ValidateV2()`. The `ValidateRequest` function exists for cases where you need to actually check the `admission.Request` object.